edgelet/edgelet-http/src/util/incoming.rs

```diff
@@ -34,6 +51,60 @@ impl Stream for Incoming {
                     Some((StreamSelector::Tcp(stream), IncomingSocketAddr::Tcp(addr)))
                 })
             }
+            #[cfg(unix)]
+            Incoming::Tls(ref mut listener, ref mut acceptor, ref mut connections) => {
+                // check if we have a tcp connection and if we do then kick off TLS handshake
+                // and store the future representing that operation in "connections"
+                if let Err(err) = listener.poll_accept().map(|val| {
+                    if let Async::Ready((tcp_stream, addr)) = val {
+                        connections
+                            .lock()
+                            .expect("Unable to lock the connection mutex")
+                            .push((acceptor.accept(tcp_stream), IncomingSocketAddr::Tcp(addr)));
+                    }
+                }) {
+                    return Err(err);
+                }
+
+                // Take a read lock on our pending connection list returning a tuple representing an index
+                // + the state on the poll
+                let mut connections = connections
+                    .lock()
+                    .expect("Unable to lock the connection mutex");
+                let val = connections
+                    .iter_mut()
+                    .map(|(fut, _)| fut.poll())
+                    .enumerate()
+                    .find(|(_, result)| match result {
+                        Ok(v) => v.is_ready(),
+                        Err(_) => true,
+                    });
```

Comment:
So, lines 74-81 is basically doing the following:

>_From the list of on-going TLS handshakes, find the first one that has either completed successfully or has failed and return a tuple containing its index and state._

Maybe we should make the comment above (on line 69) a bit more descriptive to indicate this. This code looks a little bit hard to parse otherwise.